chore(tooling): add CLAUDE.md (#1749)
ethereum/execution-spec-tests: HUMANS.md

# HUMANS.md - Working with Claude and LLMs in Ethereum Execution Spec Tests

This guide helps human developers understand the dependencies and get the most out of Claude and other LLMs when working with this codebase.

## 🤖 Why This Repository Has LLM Support

Humans are faster when they use LLMs correctly.

## 📋 Required Dependencies for LLM-Assisted Development

### Requirements

#### LLM Context File

- **CLAUDE.md** - Primary LLM guidance (keep up-to-date). Use `#memorize` in Claude to update with new info.

### Recommended Available Tooling (for use with Claude)

1. **GitHub CLI**: `gh` for PR and issue management.
2. **ripgrep**: `rg` for fast code searching.
3. **jq**: For JSON analysis and EVM trace debugging.
4. **markdownlint-cli**: To verify markdown files (CI enforced).

> 5. **VS Code**: With recommended extensions (see docs/getting_started/setup_vs_code.md). Run `claude` in VS Code for the best results.

Reviewer: Add link for easier navigation

```suggestion
5. **VS Code**: With recommended extensions ([setup guide](docs/getting_started/setup_vs_code.md)). Run `claude` in VS Code for the best results.
```

## 🎯 Getting the Best Results from Claude

### 1. Provide Relevant Context

**Always mention:**

- What you're trying to accomplish.
- Which part of the codebase you're working on (`tests/`, `src/`, `docs/`).
- Any error messages or specific issues you're encountering.

**Example - Good Context**:

> > "I'm writing a new test for EIP-7702 in `tests/prague/eip7702_set_code_tx/`. The test should verify that delegation target validation works correctly. I'm getting a `Fill failure` error when running `uv run fill --fork=Prague path/to/test.py`"

Reviewer: This is very much a nit, but I did a quick search for `Fill failure` and don't see this in the code base. Is there a more specific thing we can look for here? Since this is in code blocks (backticks), it could be better to either describe "I'm getting an error filling tests when running..." or maybe replace with the actual error in backticks? Thoughts there?

> **Example - Good Context**:
> > "I'm writing a new test for EIP-7702 in `tests/prague/eip7702_set_code_tx/`. The test should verify that delegation target validation works correctly. I'm getting a `Fill failure` error when running `uv run fill --fork=Prague path/to/test.py`"

Author: ```suggestion
**Example - Good Context**:
> "I'm writing a new test for EIP-7702 in `tests/prague/eip7702_set_code_tx/`. The test should verify that delegation target validation works correctly. The test fails to fill when running `uv run fill --fork=Prague path/to/test.py`"
```

**Example - Poor Context**:
> "My test isn't working"

### 2. Reference Key Documentation

When asking Claude for help, mention which documentation you've already checked:

- "I've read the test patterns in CLAUDE.md but...".
- "According to the debugging section in CLAUDE.md...".
- "The CONTRIBUTING.md mentions X, but I need help with Y...".

### 3. Share Specific Commands and Output

Claude works best with concrete information:

```bash
# Share the exact command you ran

> uv run fill --fork=Cancun tests/cancun/eip4844_blobs/test_blob_txs.py --clean -v

Reviewer 2: ```suggestion
uv run fill --fork=Prague tests/cancun/eip4844_blobs/test_blob_txs.py --clean -v
```
